scripts/build-zephyr.sh

```diff
@@ -0,0 +1,100 @@
+#!/bin/bash
+
+if [ "`whoami`" = "root" ]
+then
+	echo "Running the script as root is not permitted"
+	exit 1
+fi
+
+CALLED=$_
+[[ "${BASH_SOURCE[0]}" != "${0}" ]] && SOURCED=1 || SOURCED=0
+
+SCRIPT_SRC=$(realpath ${BASH_SOURCE[0]})
+SCRIPT_DIR=$(dirname $SCRIPT_SRC)
+TOP_DIR=$(realpath $SCRIPT_DIR/..)
+
+if [ $SOURCED = 1 ]; then
+	echo "You must run this script, rather then try to source it."
+	echo "$SCRIPT_SRC"
+	exit 1
+fi
+
+if [ -z "$HDMI2USB_ENV" ]; then
+	echo "You appear to not be inside the HDMI2USB environment."
+	echo "Please enter environment with:"
+	echo "  source scripts/enter-env.sh"
+	exit 1
+fi
+
+if [ -z "$ZEPHYR_SDK_INSTALL_DIR" ]; then
+	echo "Zephyr SDK not found"
+	echo "Install Zephyr SDK and/or set ZEPHYR_SDK_INSTALL_DIR variable."
+	echo "For details, see: https://www.zephyrproject.org/developers/"
+	exit 1
+fi
+
+# Imports TARGET, PLATFORM, CPU and TARGET_BUILD_DIR from Makefile
+eval $(make env)
+make info
+
+set -x
+set -e
+
+ZEPHYR_REPO=https://github.com/zephyrproject-rtos/zephyr
+ZEPHYR_REPO_BRANCH=master
+
+case $CPU in
+	vexriscv)
+		TARGET_BOARD=arty_litex_vexriscv
+		ZEPHYR_REPO=https://github.com/antmicro/zephyr
+		ZEPHYR_REPO_BRANCH=litex-vexriscv
+		;;
+	*)
+		echo "CPU $CPU isn't supported at the moment."
+		exit 1
+		;;
+esac
+
+if [ "$FIRMWARE" != "zephyr" ]; then
+	echo "When building Zephyr you should set FIRMWARE to 'zephyr'."
+	exit 1
+fi
+
+if [ ! python -c "import yaml" ]; then
+	conda install pyyaml
+fi
+
+if [ ! python -c "import elftools" ]; then
+	conda install pyelftools
+fi
+
+# Get zephyr if needed
+ZEPHYR_SRC_DIR=$TOP_DIR/third_party/zephyr
+if [ ! -d "$ZEPHYR_SRC_DIR" ]; then
+	(
+		cd $(dirname $ZEPHYR_SRC_DIR)
+		git clone -b $ZEPHYR_REPO_BRANCH $ZEPHYR_REPO
+		cd $ZEPHYR_SRC_DIR
+		git submodule update --init
+	)
+fi
+
+ZEPHYR_APP=${ZEPHYR_APP:-subsys/shell/shell_module}
+
+cd $ZEPHYR_SRC_DIR
+
+export ZEPHYR_TOOLCHAIN_VARIANT=zephyr
+source zephyr-env.sh
+
+cd samples/$ZEPHYR_APP
+mkdir build || true
+cd build
+cmake -DBOARD=$TARGET_BOARD ..
```

Comment:
It would be better if this output ended up under the `$OUTPUT_DIR` below... Maybe create a symlink?

Reply:
I switched to `west` script that allows to build outside Zephyr tree, so no moves/symlinks are needed.